File: SpringApp/src/main/java/com/example/SpringMVC/services/ProdutoService.java
package com.example.SpringMVC.services;

import com.example.SpringMVC.model.Produto;
import java.util.List;
import org.springframework.stereotype.Service;

@Service
public interface ProdutoService
{
    Produto criarProduto(Produto produto);
    Produto listarProdutoById(Long id);
    List<Produto> listarProdutos();
    Produto atualizarProduto(Produto produto);
    List<Produto> deletarProdutoById(Long id);
    List<Produto> deletarProduto(Produto produto);
}

File: angularapplication/src/app/services/produto/produto.service.ts
import { Injectable } from '@angular/core';
import { ConfigService } from '../config/config.service';
import { HttpClient } from '@angular/common/http';
import { Produto } from '../../models/produto';
import { Observable } from 'rxjs';

@Injectable()
export class ProdutoService {
  public baseUrlService = '';

  constructor(private http: HttpClient, private configService: ConfigService) {
    this.baseUrlService = configService.getUrlService() + 'api/loja'
  }

  adicionarProduto(produto: Produto): Observable<Produto> {
    return this.http.post<Produto>(this.baseUrlService + '/criar', produto);
  }

  listarProdutos(): Observable<Produto[]>  {
    return this.http.get<Produto[]>(this.baseUrlService + '/listar');
  }

  listarProdutoById(id: number): Observable<Produto> {
    return this.http.get<Produto>(this.baseUrlService + '/listar/' + id);
  }

  atualizarProduto(produto: Produto): Observable<Produto> {
    return this.http.put<Produto>(this.baseUrlService + '/atualizar/', produto);
  }

  deletarProduto(id: number): Observable<Produto[]>  {
    return this.http.delete<Produto[]>(this.baseUrlService + '/deletar/' + id);
  }
}
